src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -449,12 +449,26 @@ private static FileStream OpenStream(int fd, FileAccess access)
         /// </remarks>
         private static void ParseArgumentsIntoList(string arguments, List<string> results)
         {
-            var currentArgument = new StringBuilder();
             bool inQuotes = false;
 
             // Iterate through all of the characters in the argument string.
             for (int i = 0; i < arguments.Length; i++)
             {
+                for (; i < arguments.Length && (arguments[i] == ' ' || arguments[i] == '\t'); i++) ;
+
+                if (i == arguments.Length)
+                    break;
+
+                results.Add(GetNextArgument(arguments, ref i, ref inQuotes));
+            }
+        }
+
+        private static string GetNextArgument(string arguments, ref int i, ref bool inQuotes)
+        {
+            var currentArgument = new StringBuilder();
+
+            for (; i < arguments.Length; i++)
```

Comment:
I think the code would be clearer if you used a "while" here and increment i when needed rather than decrementing it in so many places.

In fact, consider turning all the "initialization-less" for's into while's.